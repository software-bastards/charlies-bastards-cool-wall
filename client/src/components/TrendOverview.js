import React, { Component } from "react";
import { connect } from "react-redux";
import TrendProgressBar from "./TrendProgressBar";
import "../stylesheets/TrendOverview.scss";
import calculateProgressBarPercent from "../helper/calculateProgressBarPercent";

export class TrendOverview extends Component {
  render() {
    return (
      <div className="trendBar--wrapper" data-test="component-trendoverview">
        
        {this.props.combined_votes
          ? this.props.combined_votes.map((data, index) => (
              <div key={index} className="trendBar--content_wrapper">
                <div className="trendBar--content_name">
                  {" "}
                  {data.tech_list.name}{" "}
                </div>
                <div className="trendBar--progressbar">
                  <TrendProgressBar
                    percentUncool={calculateProgressBarPercent(
                      data,
                      data.uncool_votes
                    )}
                    percentCool={calculateProgressBarPercent(
                      data,
                      data.cool_votes
                    )}
                    percentSubZero={calculateProgressBarPercent(
                      data,
                      data.subzero_votes
                    )}
                  />
                </div>
              </div>
            ))
          : null}
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    combined_votes: state.votes.votes,
  };
};

export default connect(mapStateToProps)(TrendOverview);
