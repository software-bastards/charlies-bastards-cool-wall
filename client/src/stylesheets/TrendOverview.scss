@import "../stylesheets/global.scss";

.trendBar--wrapper {
  font-size: 13px;
  width: 100%;
  margin-top: 4%;
  margin-bottom: 4%;
}

.trendBar--content_wrapper {
  padding: 2px;
  width: 94%;
  flex-direction: row;
  display: flex;
  box-shadow: $box-shadow-dark-blue;
  border-radius: $border-radius;
  align-items: center;
  margin: $margin;
}

.trendBar--content_name {

  width: 20%;
  display: flex;
  padding-left: 4%;
}

.trendBar--progressbar {
  width: 80%;
  display: flex;
}
