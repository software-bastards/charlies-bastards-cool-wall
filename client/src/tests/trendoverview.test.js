import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import TrendOverview from "../components/TrendOverview";

const mockStore = configureStore();
const store = mockStore({

    subzero_votes:10, 
    cool_votes: 5, 
    uncool_votes: 3, 
    tech_list: {name: "Express"}
});

const props = {subzero_votes:10, 
    cool_votes: 5, 
    uncool_votes: 3, 
    tech_list: {name: "Express"}};

describe('component-trendoverview', () => {
    it('should render correctly with props and redux store', () => {
        const wrapper = shallow(
            <Provider store={store}>
                <TrendOverview {...props} />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
});