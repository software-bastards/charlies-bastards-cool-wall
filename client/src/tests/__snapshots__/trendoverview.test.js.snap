// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component-trendoverview should render correctly with props and redux store 1`] = `ShallowWrapper {}`;
